server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyRequestBody.scala

```diff
@@ -36,11 +40,74 @@ class NettyRequestBody[F[_]](createFile: ServerRequest => F[TapirFile])(implicit
             Files.write(file.toPath, requestContentAsByteArray)
             RawValue(FileRange(file), Seq(FileRange(file)))
           })
-      case _: RawBodyType.MultipartBody => ???
+      case m: RawBodyType.MultipartBody =>
+        monadError
+          .unit(new HttpPostMultipartRequestDecoder(nettyRequest(serverRequest)))
+          .flatMap(decoder => {
+            getParts(serverRequest, m, decoder)
+              .sequence()
+              .map(RawValue.fromParts)
+              .map(a => {
+                decoder.destroy()
+                a.asInstanceOf[RawValue[RAW]]
+              })
+          })
     }
   }
 
-  override def toStream(serverRequest: ServerRequest): streams.BinaryStream = throw new UnsupportedOperationException()
-
   private def nettyRequest(serverRequest: ServerRequest): FullHttpRequest = serverRequest.underlying.asInstanceOf[FullHttpRequest]
+
+  private def getParts(
+      serverRequest: ServerRequest,
+      m: RawBodyType.MultipartBody,
+      decoder: HttpPostMultipartRequestDecoder
+  ): List[F[Part[Any]]] = {
+    decoder.getBodyHttpDatas.asScala
+      .flatMap(httpData =>
+        httpData.getHttpDataType match {
+          case HttpDataType.Attribute =>
+            m.partType(httpData.getName).map(c => toPart(serverRequest, c, httpData.asInstanceOf[Attribute], None)).toList
+          case HttpDataType.FileUpload =>
+            m.partType(httpData.getName)
+              .map(c => {
+                val upload = httpData.asInstanceOf[FileUpload]
+                toPart(serverRequest, c, upload, Some(upload.getContentType))
+              })
+              .toList
+          case HttpDataType.InternalAttribute => throw new UnsupportedOperationException("DataType not supported")
+        }
+      )
+      .toList
+  }
+
+  private def toPart(
+      serverRequest: ServerRequest,
+      m: RawBodyType[_],
+      upload: HttpData,
+      contentType: Option[String]
+  ): F[Part[Any]] = {
+    val mediaType = contentType.flatMap(c => MediaType.parse(c).toOption)
+    m match {
+      case RawBodyType.StringBody(charset) =>
+        monadError.unit(Part(name = upload.getName, body = upload.getString(charset), contentType = mediaType))
+      case RawBodyType.ByteBufferBody => monadError.unit(Part(name = upload.getName, body = upload.content(), contentType = mediaType))
+      case RawBodyType.InputStreamBody =>
+        monadError.unit(Part(name = upload.getName, body = new ByteBufInputStream(upload.content()), contentType = mediaType))
+      case RawBodyType.ByteArrayBody => monadError.unit(Part(name = upload.getName, body = upload.get(), contentType = mediaType))
+      case RawBodyType.FileBody =>
+        createFile(serverRequest)
+          .map(file => {
+            upload.renameTo(file)
```

Comment:
this definitely should be in a `monad.blocking` call

or maybe, there's a way to configure how Netty creates temporary files? We could use then whatever is provided in the options and bypass this step entirely